src/services/download/sentry.rs

```diff
@@ -219,37 +219,60 @@ impl SentryDownloader {
         file_source: SentryObjectFileSource,
         destination: PathBuf,
     ) -> Result<DownloadStatus, DownloadError> {
-        let download_url = file_source.url();
-        log::debug!("Fetching debug file from {}", download_url);
-        let result = future_utils::retry(|| {
-            self.client
-                .get(download_url.clone())
-                .header("User-Agent", USER_AGENT)
-                .header(
-                    "Authorization",
-                    format!("Bearer {}", &file_source.source.token),
-                )
-                .send()
-        });
-
-        match result.await {
+        match future_utils::retry(|| {
+            self.download_source_once(file_source.clone(), destination.clone())
+        })
+        .await
+        {
+            Ok(status) => {
+                log::debug!(
+                    "Fetched debug file from {}: {:?}",
+                    file_source.url(),
+                    status
+                );
+                Ok(status)
+            }
+            Err(err) => {
+                log::debug!(
+                    "Failed to fetch debug file from {}: {}",
+                    file_source.url(),
+                    err
+                );
+                Err(err)
+            }
+        }
+    }
+
+    async fn download_source_once(
+        &self,
+        source: SentryObjectFileSource,
+        destination: PathBuf,
+    ) -> Result<DownloadStatus, DownloadError> {
+        match self
+            .client
+            .get(source.url())
+            .header("User-Agent", USER_AGENT)
+            .bearer_auth(&source.source.token)
+            .send()
+            .await
```

Comment:
_nit_: In previous code, this was assigned to a variable, which makes it nicer to read. Same, and more importantly, above with the retry block. We generally use this style more:

```rust
        let retries = future_utils::retry(|| {
            self.download_source_once(file_source.clone(), destination.clone())
        });

        match retries.await { ... }
```

Reply:
i always have mixed feelings about these.  Especially when variable names like `result` appear.  And also about using un-awaited futures in an async fn.  I could get used to it I guess (though hey, can you write custom clippy lints?  because that should be linted if we actually care)

I do dislike the match statement in one expression when it closes the braces/brackets on that line though like is the case on line 225.  And I guess this case is somewhat consistent if you start doing that there.